packages/rpc-spec-types/src/overloads.ts

```diff
@@ -1,7 +1,448 @@
 export type OverloadImplementations<T, U extends keyof T> = Overloads<T[U]>;
-export type Overloads<T> = Overloads24<T>;
+export type Overloads<T> = Overloads31<T>;
```

Comment:
So yeah, I increased this to 31 to cover all the new cases. Turns out, TypeScript is sort of fast enough now. I timed a full repo `pnpm turbo test:typecheck` and it increased from 28s to 32s. I could not perceive a slowdown in terms of interactive autocomplete.